### ISSUE TITLE
unused import warnings in generated listener files

### ISSUE THREAD

**Issue by arifogel** [antlr/antlr4#838]
Every LIstener and BaseListener java file generated in my project contains the line:
  import org.antlr.v4.runtime.misc.NotNull;
without any references to NotNull appearing in the file.
This generates a ton of warnings in Eclipse.
Here is an example generated method in one of the BaseListener files:
	/**
	 * {@inheritDoc}
	 *
	 * \<p\>The default implementation does nothing.\</p\>
	 */
	@Override public void enterBraced_clause(JuniperParser.Braced_clauseContext ctx) { }

I noted that in some forum posts, the function signatures contain @NotNull after the open parenthesis. This is not occurring in my generated classes. Any ideas?
I just switched to antlr 4.5 form antlr 4.4, and I don't recall this happening before.
EDIT:
I should point out that this occurs using Java 1.8
My project can be tested by checking out https://github.com/arifogel/batfish, then sourcing batfish/tools/batfish_functions.sh, and running batfish_build in a POSIX shell. After building once, the project in batfish/projects/batfish can be opened in Eclipse.

**Comment by parrt** [comment]
Yeah, I think we let this slip through and our latest master branch should have the fix; sorry we don't have a release yet.

**Comment by arifogel** [comment]
I tried checking out antlr and following the build instructions, but it craps out. I ended up installing the latest version of mono, and changing some '4.5' literals and filenames to '4.5.1', but I got stuck. During the tests the build fails because it cannot find the NotNull class in the org.antlr4....misc package (I forgot exactly; I already deleted everything). I grepped the whole repository for NotNull, and couldn't find its definition anywhere nor any binary files containing it. So I think it's safe to say at this point that the repo is not in a buildable state.
TL;DR: Is there any way to get a usable 4.5.1 jar so I don't have to sift through all these warnings as I work? I'd rather not write a workaround to achieve this.

**Comment by ericvergnaud** [comment]
How does the 'latest version of mono' relate to this issue?
Until 4.5.1 is released, you can either edit the file org ▸ antlr ▸ v4 ▸ tool ▸ templates ▸ codegen ▸ Java.stg and remove the offending import, or add antlr annotations to your build path.
4.5.x is built using bild.py

**Comment by arifogel** [comment]
You can ignore the mono comment; it's not an issue with the repo, but with Ubuntu Trusty's outdated mono version.

I see the file you mentioned in the 4.5 release jar. If you're saying that I can simply change that file and repack the jar, then that solves my problem until the next release.

I'm not sure what you mean by 'add antlr annotations to your build path'. Which problem is that supposed to fix? My inability to build antlr 4.5.1 from source? If so, where are the antlr annotations? I ask because e.g. the word 'NotNull' does not appear in any files in any of the antlr 4 git repositories I pulled except by reference; it is never defined anywhere.

**Comment by parrt** [comment]
The 4.5.1 source should not have any references to NotNull in Java or .stg files. hopefully ;)

**Comment by parrt** [comment]
Hi. a few NotNull were left in. i just verified bild.py now works (on my mac) with latest of all antlr4-* repos.

**Comment by david-hall** [comment]
I just upgraded from 4.5 to 4.5.3 and haven't changed anything else in my existing project. 
The Antlr generated MyParserListener.java and MyParserBaseListener.Java files both contain "import org.antlr.v4.runtime.misc.NotNull;" 
Not only is the import unused, but now it's also deprecated, so additional warnings are generated for each file:
warning: [deprecation] NotNull in org.antlr.v4.runtime.misc has been deprecated
import org.antlr.v4.runtime.misc.NotNull;

**Comment by david-hall** [comment]
My mistake. A clean build with 4.5.3 seems to have fixed the problem.
